drivers/gpio/gpio_hpmicro.c

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright (c) 2022 HPMicro
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#define DT_DRV_COMPAT hpmicro_hpm_gpio
+
+#include <errno.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/irq.h>
+#include <soc.h>
+#include <hpm_gpio_drv.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+
+struct gpio_hpm_config {
+	struct gpio_driver_config common;
+	GPIO_Type *gpio_base;
+	uint32_t  port_base;
+#ifdef CONFIG_PINCTRL
+	const struct pinctrl_dev_config *pincfg;
+#endif /* CONFIG_PINCTRL */
```

Comment:
why is a GPIO driver using pinctrl ?

Reply:
Most of the pins are `gpio` functions by default, but some pins are other functions by default and need to be modified using `pinctrl`, and secondly, drive capability, pull-up and pull-down, etc. also need to be configured by `pinctrl`.